src/libraries/Microsoft.PowerFx.Core/Syntax/Visitors/ListFunctionVisitor.cs

```diff
@@ -2,32 +2,53 @@
 // Licensed under the MIT license.
 
 using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Texl;
 
 namespace Microsoft.PowerFx.Syntax
 {
     internal class ListFunctionVisitor : IdentityTexlVisitor
     {
         // FullName --> Name 
         // Use Fullname as key because it's unique. 
-        private readonly HashSet<string> _functionNames = new HashSet<string>();
+        private readonly HashSet<string> _functionNames = new HashSet<string>();
+        private readonly Dictionary<string, string> _unknownFunctionNames = new Dictionary<string, string>();
+        private readonly bool _annonymizeUnknownPublicFunctions;
 
-        public static IEnumerable<string> Run(ParseResult parse)
+        public static IEnumerable<string> Run(ParseResult parse, bool annonymizeUnknownPublicFunctions = false)
         {
-            var visitor = new ListFunctionVisitor();
+            var visitor = new ListFunctionVisitor(annonymizeUnknownPublicFunctions);
             parse.Root.Accept(visitor);
             return visitor._functionNames;
+        }
+
+        private ListFunctionVisitor(bool annonymizeUnknownPublicFunctions)
+        {
+            _annonymizeUnknownPublicFunctions = annonymizeUnknownPublicFunctions;
         }
 
         public override bool PreVisit(CallNode node)
         {
-            var hasNamespace = node.Head.Namespace.Length > 0;
-
-            var name = node.Head.Name;
-            var fullName = hasNamespace ?
-                    node.Head.Namespace + "." + name :
-                    name;
-
-            _functionNames.Add(fullName);
+            var hasNamespace = node.Head.Namespace.Length > 0;
+
+            if (_annonymizeUnknownPublicFunctions && !BuiltinFunctionsCore.IsKnownPublicFunction(node.Head.Name))

```

Comment:
>BuiltinFunctionsCore.IsKnownPublicFunction(node.Head.Name) [](http://example.com/codeflow?start=54&length=58)

Is this check sufficient? Ie will Connector Function will also get anonymized as customFunction?

Reply:
`StructuralPrint` doesn't have this issue, because looks like it doesn't care about a difference between custom function vs connector function (because all of them are external function).

Here we are specifying that it is a CustomFunction. If instead CustomFunction, we were to say Function or ExternalFunction this would be fine.